File: pythonproject/service/estatistica_medias.py
from service.estatistica_base import EstatisticaBase
from typing import List

class EstatisticaMedias(EstatisticaBase):
    def calcular(self) -> List[float]:
        medias = []
        media_geral = 0.0
        
        for curso in self.cursos:
            if curso.get_candidatos():
                nota_total = sum(candidato.get_nota() for candidato in curso.get_candidatos())
                media_curso = nota_total / len(curso.get_candidatos())
            else:
                media_curso = 0.0
            medias.append(media_curso)
            media_geral += media_curso
        
        if self.cursos:
            media_geral = media_geral / len(self.cursos)
        else:
            media_geral = 0.0
            
        medias.append(media_geral)
        
        return medias

File: javaproject/src/main/java/service/EstatisticaMedias.java
package service;

import model.*;
import java.util.List;
import java.util.ArrayList;

public class EstatisticaMedias extends EstatisticaBase {
    @Override
    public List<Double> calcular() {
        List<Double> medias = new ArrayList<>();
        Double mediaGeral = 0.0;
        
        for(Curso curso : cursos) {
            Double notaTotal = 0.0;
            for(Pessoa candidato : curso.getCandidatos()) {
                notaTotal += candidato.getNota();
            }
            Double mediaCurso = notaTotal / curso.getCandidatos().size();
            medias.add(mediaCurso);
            mediaGeral += mediaCurso;
        }
        
        mediaGeral = mediaGeral / cursos.size();
        medias.add(mediaGeral);
        
        return medias;
    }
}